node/src/p2p.rs

```diff
@@ -39,6 +43,15 @@ pub enum P2pError {
 
     #[error("Worker died")]
     WorkerDied,
+
+    #[error("Not connected to any peers")]
+    NoPeers,
+
+    #[error("Header not found")]
+    HeaderNotFound,
+
+    #[error("Unsupported header response")]
+    UnsupportedHeaderResponse,
```

Comment:
maybe more sth like `ExchangeHeaderInvalid`? it combines the exchange protocol and `Invalid` status code